outrank/algorithms/synthetic_data_generators/cc_generator.py

```diff
@@ -0,0 +1,849 @@
+                decision_function = lambda x: np.sum(2 * x + 3, axis=1)
+            elif class_relation == 'nonlinear':
+                decision_function = lambda x: np.sum(k * np.sin(x) + k * np.cos(x), axis=1)
+            elif class_relation == 'cluster':
+                decision_function = None
+        else:
+            class_relation = str(decision_function.__name__)
+
+        y = []
+        if decision_function is not None:
+            if n > 2:
+                if type(p) != list:
+                    p = 1 / n
+                    percentiles = [p * 100]
+                    for i in range(1, n - 1):
+                        percentiles.append(percentiles[i - 1] + (p * 100))
+
+                    decision_boundary = decision_function(X)
+                    p_points = np.percentile(decision_boundary, percentiles)
+
+                    y = np.zeros_like(decision_boundary, dtype=int)
+                    for p_point in p_points:
+                        y += (decision_boundary > p_point)
+                else:
+                    decision_boundary = decision_function(X)
+                    percentiles = [x * 100 for x in p]
+
+                    for i in range(1, len(percentiles) - 1):
+                        percentiles[i] += percentiles[i - 1]
+
+                    percentiles.insert(0, 0)
+                    percentiles.pop()
+                    print(percentiles)
+
+                    p_points = np.percentile(decision_boundary, percentiles)
+                    print(p_points)
+
+                    y = np.zeros_like(decision_boundary, dtype=int)
+                    for i in range(1, n):
+                        p_point = p_points[i]
+                        for j in range(len(decision_boundary)):
+                            if decision_boundary[j] > p_point:
+                                y[j] += 1
+            else:
+                decision_boundary = decision_function(X)
+                p_point = np.percentile(decision_boundary, p * 100)
+                y = np.where(decision_boundary > p_point, 1, 0)
+        else:
+            if p == 0.5:
+                p = 1.0
+            else:
+                p = [p, 1 - p]
+            y = self._cluster_data(X, n, p=p, balance=balance)
+
+        self.dataset_info.update({
+            'labels': {
+                'class_relation': class_relation,
+                'n_class': n,
+            },
+        })
+
+        return y
+
+    def _cluster_data(
+        self,
+        X: ArrayLike,
+        n: int,
+        p: float | list[float] | ArrayLike | None = 1.0,
+        balance: bool = False,
+    ) -> np.ndarray:
+        """
+        Cluster data using kmeans
+        :param X: dataset
+        :param n: number of clusters
+        :param p: class distribution
+        :param balance: balance the clusters according to p
+        :return: array of labels, corresponding to dataset X
+        """
+
+        kmeans = KMeans(n_clusters=n)
```

Comment:
Seed for kmeans set?

Reply:
Added ```random_state``` parameter for setting clustering seed.